Add new domain security screen

hambai opened the pull request:
#### Changes proposed in this Pull Request

* A new feature of the domain status redesign is the security screen that shows if your domain has a valid SSL certificate

Here's how the nav item looks:
<img width="741" alt="Screenshot 2020-05-06 at 17 16 40" src="https://user-images.githubusercontent.com/1355045/81188146-a6e77900-8fbd-11ea-8e2d-185994ec1697.png">
<img width="739" alt="Screenshot 2020-05-06 at 17 16 15" src="https://user-images.githubusercontent.com/1355045/81188155-ab139680-8fbd-11ea-9a3b-e4d395e03c8f.png">
<img width="767" alt="Screenshot 2020-05-06 at 17 13 45" src="https://user-images.githubusercontent.com/1355045/81188175-aea71d80-8fbd-11ea-9877-1924d278b9f5.png">

And here are the actual screens:
<img width="750" alt="Screenshot 2020-05-06 at 17 14 17" src="https://user-images.githubusercontent.com/1355045/81188254-c5e60b00-8fbd-11ea-9574-358853819f61.png">
<img width="750" alt="Screenshot 2020-05-06 at 17 15 54" src="https://user-images.githubusercontent.com/1355045/81188289-cf6f7300-8fbd-11ea-97d8-56fa099a84a6.png">
<img width="741" alt="Screenshot 2020-05-07 at 15 41 55" src="https://user-images.githubusercontent.com/1355045/81295937-d78ce880-9079-11ea-8e3d-397eda084d88.png">

depends on : D42936-code

#### Testing instructions

* Open a domain with certificate that's still pending or one that is active. Also test with expired domain.

a8ci18n commented on client/my-sites/domains/domain-management/security/index.jsx:
```diff
@@ -0,0 +1,191 @@
+	header() {
+		return (
+			<Header onClick={ this.back } selectedDomainName={ this.props.selectedDomainName }>
+				{ this.props.translate( 'Domain security' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Review your domain security' )` *ES Score: 7*

a8ci18n commented on client/my-sites/domains/domain-management/security/index.jsx:
```diff
@@ -0,0 +1,191 @@
+		return (
+			<React.Fragment>
+				<CompactCard className="security__header">
+					<span>{ translate( 'HTTPS encryption' ) }</span>
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( 'HTTPS encryption: on' )` *ES Score: 9*

a8ci18n commented on client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx:
```diff
@@ -349,16 +359,50 @@ class DomainManagementNavigationEnhanced extends React.Component {
+				} );
+				break;
+			case sslStatuses.SSL_DISABLED:
+				sslStatusHuman = translate( 'disabled', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **36** times:
`translate( 'Disabled.' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-domain-security-screen)

fditrapani commented:
For some reason, I'm not seeing the option in my menu. I'm proxied to the store and have the patch enabled. Any ideas why it's not showing up?

From a design perspective, the screenshots look good except for the contact support it — it feels like a bit much. Can we change the big block of copy with a button instead? Then when you click on that, we can open up the support widget like we're doing in this PR: https://github.com/Automattic/wp-calypso/pull/41811

Here's a screenshot for reference:

![image](https://user-images.githubusercontent.com/6981253/81213001-beb20380-8fa3-11ea-8961-a75c4373a36a.png)

After not looking at this, for a while, I also made some minor comments on copy updates.

fditrapani commented on client/my-sites/domains/domain-management/security/index.jsx:
```diff
@@ -0,0 +1,197 @@
+				<React.Fragment>
+					<p>
+						{ translate(
+							'Due to some changes to your domain, we need to generate a new SSL certificate to activate your HTTPS encryption. This process should only take a couple hours at most so if you’re running into delays please let us know so we can help you out.'
```
Can we update

```Due to some changes to your domain, we need to generate a new SSL certificate to activate your HTTPS encryption. This process should only take a couple hours at most so if you’re running into delays please let us know so we can help you out.```

to:
```Due to some changes to your domain, we need to generate a new SSL certificate to activate your HTTPS encryption. This process should only take a couple hours at most. If you’re running into delays please let us know so we can help you out.```

fditrapani commented on client/my-sites/domains/domain-management/security/index.jsx:
```diff
@@ -0,0 +1,197 @@
+				<ul>
+					{ [
+						translate( 'Trust indicators that reassure your visitors your site is safe ' ),
+						translate( 'Secure data transmission of data sent through forms' ),
```
Can we update this to "Secure data transmission for all your forms

fditrapani commented on client/my-sites/domains/domain-management/security/index.jsx:
```diff
@@ -0,0 +1,197 @@
+				</CompactCard>
+				{ sslStatuses.SSL_ACTIVE === sslStatus && (
+					<VerticalNav>
+						<VerticalNavItem path={ ECOMMERCE } external>
```
Do we have any events firing for these links? I'd love to keep an eye on them to see if they trigger any upgrades.

fditrapani commented on client/my-sites/domains/domain-management/security/style.scss:
```diff
@@ -0,0 +1,33 @@
+			}
+
+			&.disabled svg {
+				fill: var( --studio-pink-50 );
```
Can we change this to `--color-error` so that it'll carry over to the different Calypso themes appropriately?

fditrapani commented on client/my-sites/domains/domain-management/security/style.scss:
```diff
@@ -0,0 +1,33 @@
+				fill: var( --studio-pink-50 );
+			}
+			&.active svg {
+				fill: var( --studio-green-40 );
```
Can we change this to `--color-success` so that it'll carry over to the different Calypso themes appropriately?

hambai commented:
@fditrapani Okay, updated the PR to have a simple "Contact support" button, updated the copy and SCSS vars and added `calypso_domain_security_learn_more_click` to track the learn more clicks. Did you apply the backend diff to your sandbox to test this?

fditrapani commented:
Cool, thanks for the updates. 

Yeah, I'm sandboxed and applied the diff. (does it make a difference if I'm sandboxed to the store)? I still don't see the option.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3554056

Thank you @hambai for including a screenshot in the description! This is really helpful for our translators.
